src/components/widget/line/widget-chart-line.component.ts

```diff
@@ -54,47 +56,91 @@ const WidgetChartLineComponent: ng.IComponentOptions = {
           $scope.$on('onWidgetResize', this.onResize.bind(this));
 
         } else {
-          this.gvChartLine.setAttribute('series', JSON.stringify([]));
+          this.gvChartLine.setAttribute('series', JSON.stringify([{}]));
         }
       }
     };
 
     this.prepareData = (data) => {
       let i;
+      let timestamp = data.timestamp;
+      this.events = data.events && data.events.content;
+
       data.values.forEach( (value, idx) => {
-        value.buckets.forEach((bucket) => {
-          if (bucket) {
-            i++;
-            let lineColor = ChartService.colorByBucket[i % ChartService.colorByBucket.length];
-            let bgColor = ChartService.bgColorByBucket[i % ChartService.bgColorByBucket.length];
-            let isFieldRequest = this.parent.widget.chart.request.aggs.split('%3B')[idx].includes('field:');
-            let query = this.parent.widget.chart.request.query;
-            if (bucket.name === '1' || bucket.name.match('^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$')) {
-              isFieldRequest = false;
-            }
-            let label = this.parent.widget.chart.labels ? this.parent.widget.chart.labels[idx] : '';
-            if (!label || (value.metadata && value.metadata[bucket.name])) {
-              label = value.metadata[bucket.name].name;
+        i++;
```

Comment:
It was there before but I think we can improve it: 
This `i++` doesn't work as `i` has never been initialized, see L65.
I would remove this `i` and use `idx` variable directly in L71 L72, it will simplify the code and also fix the line + bg Color which are currently fallbacking to HighChart default values.

Reply:
we keep highcharts colors so this code is deleted